static/src/javascripts/projects/common/modules/identity/manage-account.js

```diff
@@ -37,17 +39,54 @@ const submitPartialFormStatus = (
     });
 };
 
+const getNewsletterHtmlPreferenceFromElement = (
+    originalEl: HTMLElement
+): Promise<string> =>
+    fastdom.read(() => {
+        const closestFormEl: ?Element = originalEl.closest('form');
+
+        if (!closestFormEl) throw Error(ERR_HTML_PREF_NOT_FOUND);
+
+        const checkboxEl: ?HTMLElement = closestFormEl.querySelector(
+            '[name="htmlPreference"]:checked'
+        );
+        const inputEl: ?HTMLElement = closestFormEl.querySelector(
```

Comment:
Would it be better to query for all `[name="htmlPreference"]` and then filter out `:checked`, in order to save one query?

Reply:
might get a bit messy, i'm gonna try my hand at it but the code might end up ugly-er bc ultimately i want it to work on at least radio buttons and hidden inputs